packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -91,12 +94,12 @@ export const Alert: React.FunctionComponent<AlertProps> = ({
       setIsTooltipVisible(showTooltip);
     }
   }, [titleRef, truncateTitle, isTooltipVisible]);
-  const customClassName = css(
-    styles.alert,
-    isInline && styles.modifiers.inline,
-    variant !== AlertVariant.default && styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-    className
-  );
+  React.useEffect(() => {
+    timeout = timeout === true ? 8000 : Number(timeout);
+    if (timeout > 0) {
+      setTimeout(() => setTimedOut(true), timeout);
```

Comment:
You need to clear the timeout on unmount otherwise you'll end up updating state on an unmounted component if the component is unmounted before the timeout completes.